File: sdks/guides/ruby/integrations/ocr_tesseract.rb
# frozen_string_literal: true

# Third-party integration - requires: bundle add rtesseract; install Tesseract binary
# See: /guides/document-processing

require 'rtesseract'

# > Tesseract usage
def parse_document(content)
  RTesseract.new(nil, data: content).to_s
end
# !!

File: examples/python/guides/integrations/ocr_tesseract.py
# Third-party integration example - requires: pip install pytesseract; install Tesseract binary
# See: /guides/document-processing

import io

import pytesseract
from PIL import Image


# > Tesseract usage
def parse_document(content: bytes) -> str:
    img = Image.open(io.BytesIO(content))
    return str(pytesseract.image_to_string(img))
